src/fs/spec/sync.spec.ts

```diff
@@ -0,0 +1,46 @@
+import path from 'path'
+import {
+  generateProgramToGetRemoteHash,
+  generateProgramToSyncHashDiff,
+  findResourcesNotPresentLocally
+} from '../sync'
+import { getHash } from '../hash'
+import { createFile, readFile } from '../../file'
+
+describe('generateProgramToGetRemoteHash', () => {
```

Comment:
Presumably this test will fail if there is even a  minor change to the source file in sasjs/core, eg this one?  https://github.com/sasjs/utils/pull/208/files#diff-e132e585f51871720b746bd1fae142ccb61b286bfc80ada6e2adfd4a1e4ee46fR69

Perhaps we should check for the existence of key components (eg lines beginning with `%macro mp_hashdirectory` and `%macro mp_jsonout`) rather than doing an exact match at file level against a static copy of a sasjs/core component?

Reply:
yes, Good Point.